crates/bevy_core_pipeline/src/motion_blur/pipeline.rs

```diff
@@ -0,0 +1,174 @@
+}
+
+impl MotionBlurPipeline {
+    pub(crate) fn new(render_device: &RenderDevice) -> Self {
+        let mb_layout = &BindGroupLayoutEntries::sequential(
+            ShaderStages::FRAGMENT,
+            (
+                // View target (read)
+                texture_2d(TextureSampleType::Float { filterable: true }),
+                // Motion Vectors
+                texture_2d(TextureSampleType::Float { filterable: true }),
+                // Depth
+                texture_depth_2d(),
+                // Linear Sampler
+                sampler(SamplerBindingType::Filtering),
+                // Motion blur settings uniform input
+                uniform_buffer_sized(false, Some(MotionBlur::min_size())),
+                // Globals uniform input
+                uniform_buffer_sized(false, Some(GlobalsUniform::min_size())),
+            ),
+        );
+
+        let mb_layout_msaa = &BindGroupLayoutEntries::sequential(
+            ShaderStages::FRAGMENT,
+            (
+                // View target (read)
+                texture_2d(TextureSampleType::Float { filterable: true }),
+                // Motion Vectors
+                texture_2d_multisampled(TextureSampleType::Float { filterable: false }),
+                // Depth
+                texture_depth_2d_multisampled(),
+                // Linear Sampler
+                sampler(SamplerBindingType::Filtering),
+                // Motion blur settings uniform input
+                uniform_buffer_sized(false, Some(MotionBlur::min_size())),
+                // Globals uniform input
+                uniform_buffer_sized(false, Some(GlobalsUniform::min_size())),
+            ),
+        );
+
+        let sampler = render_device.create_sampler(&SamplerDescriptor::default());
+        let layout = render_device.create_bind_group_layout("motion_blur_layout", mb_layout);
+        let layout_msaa =
+            render_device.create_bind_group_layout("motion_blur_layout_msaa", mb_layout_msaa);
+
+        Self {
+            sampler,
+            layout,
+            layout_msaa,
+        }
+    }
+}
+
+impl FromWorld for MotionBlurPipeline {
+    fn from_world(render_world: &mut bevy_ecs::world::World) -> Self {
+        let render_device = render_world.resource::<RenderDevice>().clone();
+        MotionBlurPipeline::new(&render_device)
+    }
+}
+
+#[derive(PartialEq, Eq, Hash, Clone, Copy)]
+pub struct MotionBlurPipelineKey {
+    hdr: bool,
+    samples: u32,
+}
+
+impl SpecializedRenderPipeline for MotionBlurPipeline {
+    type Key = MotionBlurPipelineKey;
+
+    fn specialize(&self, key: Self::Key) -> RenderPipelineDescriptor {
+        let layout = match key.samples {
+            1 => vec![self.layout.clone()],
+            _ => vec![self.layout_msaa.clone()],
+        };
+
+        let mut shader_defs = vec![];
+
+        match key.samples {
+            1 => (),
+            _ => shader_defs.push(ShaderDefVal::from("MULTISAMPLED")),
+        }
```

Comment:
Why not just `if key.samples > 1`?

Reply:
copy-pasta and a bunch of merges with main, I assume. :)